### PR TITLE
CTCP-4803: AddAnotherDepartureTransportMeans.

### PR DESCRIPTION
**Pull request by justega247**

### CODE REVIEW

File: app/models/journeyDomain/transportMeans/TransportMeansDepartureDomain.scala [hunk 1]
```diff
@@ -52,14 +64,14 @@ case class PostTransitionTransportMeansDepartureDomain(
...
 }
 
 object PostTransitionTransportMeansDepartureDomain {
 
-  def asString(identification: Option[Identification], identificationNumber: String)(implicit messages: Messages): String =
-    identification.fold(identificationNumber)(
-      value => s"${value.asString} - $identificationNumber"
+  def asString(identification: Option[Identification], identificationNumber: String, index: Index)(implicit messages: Messages): String =
+    identification.fold(s"Departure means of transport ${index.display} - $identificationNumber")(
+      value => s"Departure means of transport ${index.display} - ${value.asString} - $identificationNumber"
```
Comment (jackheslop96) [inline app/models/journeyDomain/transportMeans/TransportMeansDepartureDomain.scala:74]: Best practice to define these in message keys

File: app/models/journeyDomain/transportMeans/TransportMeansDepartureDomain.scala [hunk 2]
```diff
@@ -77,14 +89,23 @@ case class TransitionTransportMeansDepartureDomain(
...
 object TransitionTransportMeansDepartureDomain {
 
+  def asString(identification: Option[Identification], identificationNumber: Option[String], index: Index)(implicit messages: Messages): String =
+    (identification, identificationNumber) match {
+      case (Some(id), Some(idNumber)) => s"Departure means of transport ${index.display} - ${id.asString} - $idNumber"
+      case (Some(id), None)           => s"Departure means of transport ${index.display} - ${id.asString}"
+      case (None, Some(idNumber))     => s"Departure means of transport ${index.display} - $idNumber"
+      case _                          => s"Departure means of transport ${index.display}"
+    }
+
```
Comment (jackheslop96) [inline app/models/journeyDomain/transportMeans/TransportMeansDepartureDomain.scala:105]: Best practice to define these in message keys
e.g. 
```
departureTransportMeans.label.noArgs = Departure means of transport {0}
departureTransportMeans.label.oneArg = Departure means of transport {0} - {1}
departureTransportMeans.label.bothArgs = Departure means of transport {0} - {1} - {2}
```

File: app/pages/transportMeans/departure/AddDepartureTransportMeansYesNoPage.scala [hunk 3]
```diff
@@ -14,28 +14,35 @@
...
 import pages.QuestionPage
-import pages.sections.transportMeans.{DeparturesSection, TransportMeansSection}
+import pages.sections.transportMeans.DepartureSection
 import play.api.libs.json.JsPath
 import play.api.mvc.Call
 
 import scala.util.Try
 
-case object AddDepartureTransportMeansYesNoPage extends QuestionPage[Boolean] {
+case class AddDepartureTransportMeansYesNoPage(departureIndex: Index) extends QuestionPage[Boolean] {
```
Comment (jackheslop96) [inline app/pages/transportMeans/departure/AddDepartureTransportMeansYesNoPage.scala:28]: Not sure about this? Is this not the yes/no question for whether they want to add **any** departure transport means? i.e. no index

File: app/utils/cyaHelpers/transportMeans/TransportMeansCheckYourAnswersHelper.scala [hunk 4]
```diff
@@ -95,10 +96,10 @@ class TransportMeansCheckYourAnswersHelper(
     id = Some("change-transport-means-inland-mode")
   )
 
-  def addDepartureTransportMeans: Option[SummaryListRow] = getAnswerAndBuildRow[Boolean](
-    page = AddDepartureTransportMeansYesNoPage,
+  def addDepartureTransportMeans(departureIndex: Index): Option[SummaryListRow] = getAnswerAndBuildRow[Boolean](
```
Comment (jackheslop96) [inline app/utils/cyaHelpers/transportMeans/TransportMeansCheckYourAnswersHelper.scala:99]: This is a duplication of the method you've added in DepartureTransportMeansAnswersHelper? I don't think this page should have an index

File: app/models/journeyDomain/transportMeans/TransportMeansDepartureDomain.scala [hunk 5]
```diff
@@ -18,19 +18,31 @@ package models.journeyDomain.transportMeans
...
 
 sealed trait TransportMeansDepartureDomain extends JourneyDomainModel {
   val index: Index
 
   def asString(implicit messages: Messages): String
+
+  override def routeIfCompleted(userAnswers: UserAnswers, mode: Mode, stage: Stage, phase: Phase): Option[Call] = Some {
+    stage match {
+      case AccessingJourney =>
+        controllers.transportMeans.departure.routes.AddIdentificationTypeYesNoController
```
Comment (jackheslop96) [inline app/models/journeyDomain/transportMeans/TransportMeansDepartureDomain.scala:40]: Is this TODO needed? That code looks right to me

**Review by jackheslop96 — APPROVED** [review]

**Review by kenneth-rayner — APPROVED** [review]

**Pull request merged**